enhancements/namespace-auto-pruning.md

```diff
@@ -0,0 +1,170 @@
+---
+title: Namespace Auto-pruning Policies
+authors:
+  - "@bcaton85"
+reviewers:
+  - "@dmage"
+  - "@Sunandadadi"
+approvers:
+  - "@dmage"
+  - "@Sunandadadi"
+creation-date: 2023-08-15
+last-updated: 2023-08-15
+status: provisional
+---
+
+# Namespace Auto-pruning Policies
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+1. Should there be a dry run feature to test the policy?
+1. Should this only be implemented in the new UI and not the old UI?
+1. Can a user configure multiple policies per namespace?
+1. Can a user specify tags or repositories by pattern?
+1. Do we want the ability to have tiered priority for auto-prune execution?
+
+## Summary
+
+In order for a container registry to run cost effectively unused resources should be deleted to reclaim storage space. This task should be done automatically in Quay by deleting unused tags through a given policy. The policy used should be configured by the user and a variety policy options should be given. Examples of initial policies would be prune tags older than X number of days or prune tags when more than 100 are present.
+
+## Motivation
+
+Reclaiming space is essential in running a cost efficient registry. Having this done automatically within Quay removes the need for manual intervention or custom software.
+
+### Goals
+
+* User is able to provide a time based policy for auto-pruning tags within a namespace
+* User is able to provide a tag number based policy for auto-pruning tags within a namespace
```

Comment:
This would be a target number of alive tags in a namespace. Say it's configured at 10 tags but 11 exist one tag will be pruned. That tag will be pruned by the oldest creation date.